packages/storefront-lwc/src/modules/commerce/productDetail/productDetail.js

```diff
@@ -130,12 +134,44 @@ export default class ProductDetail extends LightningElement {
         return canAddToBasket(this.product, this.selectedQty);
     }
 
+    @wire(useMutation, { mutation: ADD_TO_BASKET }) addToBasket;
+
     /**
      * Add product to basket when user clicks `Add to Basket` button
      */
     addToBasketHandler() {
-        ShoppingBasket.addToBasket(this.product, this.selectedQty);
+        if (this.readyToAddToBasket) {
+            this.productId = this.pid;
+            this.quantity = parseInt(this.selectedQty);
```

Comment:
It seems weird that selectedQty could possibly be a string. I traced it down and I think it originates from `event.target.value` inside the variations component. I think it would be better to cast the quantity as an integer there of here: https://github.com/SalesforceCommerceCloud/sfcc-sample-apps/blob/056f90fa4de87234bb88f4a2a4f3a5c8bfc643a6/packages/storefront-lwc/src/modules/commerce/variations/variations.js#L173

Reply:
Good catch, fixing it